File: src/algoexpert/easy/ValidateSubsequence/ValidateSubsequence.java
import java.util.*;

class Program {
  public static boolean isValidSubsequence(List<Integer> array, List<Integer> sequence) {
    // O(n) time | O(1) space n is the length of the array
		int j=0;
		for(int i=0;i<array.size()&&j<sequence.size();i++){
			if(array.get(i)==sequence.get(j)){
				j+=1; 
			} 			
		}
    return j==sequence.size();
  }
}

File: src/algoexpert/easy/ValidateSubsequence/ValidateSubsequence.py
def isValidSubsequence(array, sequence):
    # Write your code here.
	i = 0
	j = 0
	while i < len(array) and j < len(sequence):
		if array[i] == sequence[j]:
			j += 1
		i += 1
	return j == len(sequence)
